lib/model/model_test.go

```diff
@@ -1904,48 +1904,69 @@ func TestIssue3164(t *testing.T) {
 
 func TestIssue4357(t *testing.T) {
 	db := db.OpenMemory()
-	m := NewModel(defaultConfig, protocol.LocalDeviceID, "syncthing", "dev", db, nil)
+	cfg := defaultConfig.RawCopy()
+	// Create a separate wrapper not to polute other tests.
+	wrapper := config.Wrap("/tmp/test", config.Configuration{})
+	m := NewModel(wrapper, protocol.LocalDeviceID, "syncthing", "dev", db, nil)
 	m.ServeBackground()
 	defer m.Stop()
-	cfg := m.cfg.RawCopy()
-	m.CommitConfiguration(config.Configuration{}, cfg)
+
+	// Replace kicks off CommitConfiguration in multiple routines which we
+	// have no idea when they get called, so sleep a bit :(
+	// We could make replace blocking for all routines to come back without
+	// holding the lock, but that's a bit much just to make a test.
+	replaceCfg := func(cfg config.Configuration) {
+		t.Helper()
```

Comment:
This is new in go1.9, so that may create problems for packagers (until this 1.8 worked fine for me). That's not a problem, as we require latest go, but it might be nice to mention it in the release (@calmh).

Reply:
Actually this should be removed and the comment should be adjusted.

Reply:
Yeah doit. (We build with an old compiler, but don't test with it...)